css/cssom-1/cssom-ruleOrder.html

```diff
@@ -47,49 +49,30 @@
     <div id="log"></div>
     <script>
     var stylesheets = document.styleSheets;
-    var ruletypes = {
-      1: 'Style Rule',
+
+    var expectedRuleOrder = [10, 3, 1, 1, 5, 4, 6];
+
+    var typesText = {
+      1: 'style Rule',
       3: '@import rule',
       4: 'media rule',
       5: '@font-face rule',
       6: 'page rule',
-      10: 'Namespace rule',
+      10: 'namespace rule',
       11: 'unrecognized rule'
     };
 
-    var types = {
-      's-0': '@namespace svg "http://www.w3.org/2000/svg"',
-      's-1': '@import url("main.css")',
-      's-2': 'h1',
-      's-3': 'svg|a',
-      's-4': '@font-face',
-      's-5': '@media (min-width: 200px)',
-      's-6': '@page :first'
-    }
-
-   var ruleOrder = [10, 3, 1, 1, 5, 4, 6];
-
-    var results = [];
-
-
-    for (i = 0; i < stylesheets.length; i++) {
-      if(i == 3) {
-        cssType = stylesheets[i].cssRules.length > 0 && stylesheets[i].cssRules[1] ? stylesheets[i].cssRules[1].type : 11;
-      } else {
-        cssType = stylesheets[i].cssRules.length > 0 ? stylesheets[i].cssRules[0].type : 11;
-      }
-
-        results.push([
-          ruletypes[ruleOrder[i]] + ' is expected to be ' + types['s-' + i],
-          ruletypes[cssType],
-          ruletypes[ruleOrder[i]]
-        ]);
-
-    }
-
-    generate_tests(assert_equals, results);
-
-    </script>
+   for (var i = 0; i < stylesheets.length; i++) {
+    test( function () {
+        if (i === 3) {
+          var cssType = stylesheets[i].cssRules.length > 0 && stylesheets[i].cssRules[1] ? stylesheets[i].cssRules[1].type : 11;
```

Comment:
This seems overly complex (I realise this is pre-existing!); can it not just be `stylesheets[i].cssRules.length > 1 ? stylesheets[i].cssRules[1].type : 11`?

Actually, I'm not sure why we have the conditional operator there at all. It just seems to be providing a default value so it has something to push to the old `results` array and the check for the length is only needed to make sure it doesn't throw in case it fails (which we need in the `generate_tests` case because otherwise it won't run *any* tests, which is the whole reason why it is far too fragile).

I think we can actually literally just do `stylesheets[i].cssRules[1].type` (and similar, but 0, below), and accept that'll throw if for some reason we don't get the `cssRules` we expect, but that'll be caught by the test and then it doesn't hide that problem.

As such, this seems like simplification worth doing while getting rid of `generate_tests`.